public/app/features/alerting/unified/utils/matchers.ts

```diff
@@ -11,10 +18,18 @@ const matcherOperators = [
   MatcherOperator.equal,
 ];
 
+/**
+ * Parse a single matcher, examples:
+ *  foo="bar" => { name: foo, value: bar, isRegex: false, isEqual: true }
+ *  bar!~baz => { name: bar, value: baz, isRegex: true, isEqual: false }
+ */
 export function parseMatcher(matcher: string): Matcher {
```

Comment:
Should we mark it as deprecated?

Reply:
IIRC it's still used for parsing label matchers in the search DSL and the query params for pre-filling matchers in new silences – we might want to keep that for now to prevent too much refactoring in this PR :)